libraries/Microsoft.Bot.Builder/AppInsightsLoggerMiddleware.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// Middleware for logging incoming activitites into Application Insights.
+    /// In addition, registers a service that other Application Insights components can log
+    /// telemetry.
+    /// If this component is not registered, visibility within the Bot is not logged.
+    /// </summary>
+    public class AppInsightsLoggerMiddleware : IMiddleware
+    {
+        public const string AppInsightServiceKey = "AppInsightsContext";
+        public const string BotMsgEvent = "BotMessageReceived";
+        public const string ChannelProperty = "Channel";
```

Comment:
I would suggest a separate nested, private class to throw all these App Insights consts onto so that your primary class isn't so polluted with the noise.

Reply:
If you don't do that, I also don't see any reason for these to be public... they are maybe `internal` at best so your tests can validate against them.

Reply:
I plead code review exhaustion. :) 

+1

Reply:
Nice, good feedback